### ISSUE TITLE
S6 Pure

### ISSUE THREAD

**Issue by tommyMX** [iobroker-community-adapters/ioBroker.mihome-vacuum#726]
Hi

is the s6 also working ?

thx

**Comment by dirkhe** [comment]
Did you mean roborock s6, than definitely, because I also have one

**Comment by tommyMX** [comment]
S6 Pure , yes...

but Iobroker said:

**Comment by dirkhe** [comment]
So it is not a roborock, but ijai. This will not be supported yet. We need someone, who owned a ijai device and can implement this.

**Comment by tommyMX** [comment]
Hi Dirkhe,

mhh it was called roborock S6 Pure (https://amzn.eu/d/iZgqfDy)  anyway, i have it at home right now to test it.

Also i am using mii home app and login is working over the cloud.

Token i also added.

So i have the Device at home and already tried to initialize in iobroker.

What information do you need to implent it?

**Comment by dirkhe** [comment]
I have the roborock S6, I guess you have an old model?
What you can try
edit the file ioBroker/iobroker.mihome-vacuum/main.js and add your model to the roborock list, like
'ijai.vacuum.v19': VacuumManager  on line 40
restart adapter. if than the most will work, we can add this generally, otherwise someone has to implement a new library for ijai

**Comment by tommyMX** [comment]
its from 2019....

can i edit it from iobroker the file or login to my server? maybe the better way....

i will try

is there a list which devices are fully working?

So you cant implement or make a new library?

Which model S6 have you? have you a link from amazon that i can see the right model?

**Comment by tommyMX** [comment]
but i see the S6 Pure is in the main,js but au different model name.... mhhh :-(

**Comment by dirkhe** [comment]
I'm searching for the differneces...

**Comment by tommyMX** [comment]
maybe i have the german model?

**Comment by dirkhe** [comment]
https://github.com/PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor/issues/200
Seem' to be a different supplier. I thinbk, there is a new libary needed...

**Comment by tommyMX** [comment]
what do you mean with to map all api calls?

**Comment by tommyMX** [comment]
so the factory changed and thats the problem :-( maybe for alle roborock models...not good

**Comment by tommyMX** [comment]
so i must wait? :-)

**Comment by tommyMX** [comment]
i saw you made a link to the another topic, so we will get informed when the have updates.

so the plugin code must be rewritten again ?

**Comment by dirkhe** [comment]
yes, because they try to understand the communication and than this has to be adapt to iobroker

**Comment by tommyMX** [comment]
but when the fixed it , it is easy for you to bring it to this adapter?

**Comment by tommyMX** [comment]
https://home.miot-spec.com/spec?type=urn:miot-spec-v2:device:vacuum:0000A006:ijai-v19:1

**Comment by dirkhe** [comment]
Thi is the cloud api, the current Adapter cummunicated mostly direct to the device. It could be that this will be chnged in future, but than I stop the firmware updates, because I didn't want , that everything goes to the cloud

**Comment by tommyMX** [comment]
yes i understand, i also dont like the cloud things :-(

so it looked they changed something...so from your side you cant do anything more?

when i gave you my login credentials and you can look how the app is communcating... ?

**Comment by dirkhe** [comment]
without device it makes no sense. And it is a lot of work, there I have not enough time to do this.

**Comment by tommyMX** [comment]
i think all models will have the soon the new device name...when the factory changed...

**Comment by tommyMX** [comment]
> Model ijai.vacuum.v19 not supported! Please open issue on

i can help you in anyway...let me know what i should do with my device...

**Comment by dirkhe** [comment]
Sorry, I can't help here, because of reasons above. Fork this repository and try by your self and make a PullRequest afterwards.

**Comment by tommyMX** [comment]
i have no idea how to write a plugin like that :-(

so I thought you are the developer here?  

i hinkt I must wait

**Comment by tommyMX** [comment]
@dirkhe 
Hi, I found a plugin on GitHub which is for home bridge. And that plugin knows my device.

Will this help?

https://github.com/merdok/homebridge-miot

**Comment by xXBJXx** [comment]
Version **6.1.0** was released on 2026-09-08. It contains the runtime rewritten in 6.0.0 (typed UDP/miIO errors instead of silent failures, Xiaomi login-link authentication instead of username and password, an optional map renderer), the queue and Alexa fixes from 6.0.1, and the rebuilt VIS 1 and VIS 2 widgets. It requires Node.js 22.13 or newer, js-controller 7.2.2 or newer and Admin 7.8.23 or newer and is installed from npm through Admin (installation from GitHub is disabled on purpose).

To answer the original question: the S6 Pure reports a `roborock.vacuum.*` model id, so it uses the Roborock library like the S5/S6/S7 models in the supported-devices table; basic control, history, rooms and map are available. The later part of this thread about `ijai.vacuum.v19` is tracked in #642. Closing this one as answered; if the S6 Pure does not work on 6.1.0, please open a new issue with the debug log.
